internal/service/studies/contracts.go

```diff
@@ -2,24 +2,87 @@ package studies
 
 import (
 	"context"
+	"fmt"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/rs/zerolog/log"
+	openapi "github.com/ucl-arc-tre/portal/internal/openapi/web"
 	"github.com/ucl-arc-tre/portal/internal/types"
+	"github.com/ucl-arc-tre/portal/internal/validation"
 )
 
+func (s *Service) ValidateContractMetadata(organisationSignatory string, thirdPartyName string, status string, expiryDateStr string) *openapi.ValidationError {
+	if !validation.ContractNamePattern.MatchString(organisationSignatory) {
+		return &openapi.ValidationError{
+			ErrorMessage: "Organisation signatory must be between 2 and 100 characters",
+		}
+	}
+
+	if !validation.ContractNamePattern.MatchString(thirdPartyName) {
+		return &openapi.ValidationError{
+			ErrorMessage: "Third party name must be between 2 and 100 characters",
+		}
+	}
+
+	if status != "proposed" && status != "active" && status != "expired" {
```

Comment:
better to match enum probably

Reply:
[now using typed enum](https://github.com/ucl-arc-tre/portal/pull/295/commits/26e98fc00f5f21eab8a7082d9e0d355308af5310)